### ISSUE TITLE
False positive: error  'root' is already declared in the upper scope  no-shadow

### ISSUE THREAD

**Issue by madarche** [eslint/eslint#10206]
* **ESLint Version:** 4.19.1
* **Node Version:** 8 or 9
* **npm Version:** 5.5.1 or 5.6.0

**Parser used (default, Babel-ESLint, etc.):** default

**Full configuration:** `.eslintrc.js` :

```js
module.exports = {
    "env": {
        "es6": true,
        "node": true
    },
    "extends": "eslint:recommended",
    "rules": {
        'no-shadow': ['error', {builtinGlobals: true, hoist: 'all'}]
    }
};
```

**Source code tested with ESLint:** `routes.js` :

```js
const app = require('./app');
const root = require('./controller/root');
app.route('/').get(root.sendFrontPage);
```

**Command lines used:**

```bash
npm i eslint
./node_modules/.bin/eslint .
```

**What did you expect to happen?**

No error. The `root` variable is only defined once in this unique file.

**What actually happened? Please include the actual, raw output from ESLint.**

```
3:7  error  'root' is already declared in the upper scope  no-shadow
```

**Comment by platinumazure** [comment]
Hi @madarche, thanks for the issue.

I have a reduced test case [here](https://eslint.org/demo/#eyJ0ZXh0IjoiLyogZXNsaW50LWVudiBlczYsIG5vZGUgKi9cclxuLyogZXNsaW50IG5vLXNoYWRvdzogW2Vycm9yLCB7IGJ1aWx0aW5HbG9iYWxzOiB0cnVlIH1dICovXHJcblxyXG5jb25zdCBhcHAgPSByZXF1aXJlKCcuL2FwcCcpO1xyXG5jb25zdCByb290ID0gcmVxdWlyZSgnLi9jb250cm9sbGVyL3Jvb3QnKTtcclxuYXBwLnJvdXRlKCcvJykuZ2V0KHJvb3Quc2VuZEZyb250UGFnZSk7Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo4LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19). It appears that only the `builtinGlobals: true` configuration option is needed to trigger the issue.

It looks like we intentionally [keep `root` as a global in our `node` environment](https://github.com/eslint/eslint/blob/master/conf/environments.js#L139). Not 100% sure if that's by design or not. We recently discussed removing some deprecated browser and jest globals which we had retained for backward compatibility. Not sure if we have any plans to do the same with the deprecated globals in this case.

For now, this *appears* to be working as designed since `root` is a builtin global in our `node` environment.

**Comment by madarche** [comment]
@platinumazure thanks very much for the explanation (and fast) and the test reduction!

`root` is not documented as a Node.js global in [the Node.js documentation](https://nodejs.org/api/globals.html). So it would be good that either ESLint node globals or Node.js globals documentation gets fixed. I can help, of course, with PRs once the truth is found.

**Comment by Aladdin-ADD** [comment]
seems it would be fixed by #10210

**Comment by mysticatea** [comment]
#9144 was a similar issue.

Anyway, I'm closing this issue since this has been fixed on master branch.
